### PR TITLE
Fix NPE when loading sync from cache

### PR DESCRIPTION
**Pull request by lukebarnard1**

### CODE REVIEW

File: src/sync.js [hunk 1]
```diff
@@ -528,13 +530,13 @@ SyncApi.prototype._syncFromCache = async function(savedSync) {
     };
 
     try {
-        await this._processSyncResponse(oldSyncToken, data, true);
+        await this._processSyncResponse(oldSyncEventData, data, true);
```
Comment (richvdh) [inline src/sync.js]: why aren't we just using `syncEventData` here? it seems to be what the other call to `_processSyncResponse` does?

Comment (lukebarnard1) [reply src/sync.js]: No idea, we should be using `syncEventData`.

File: src/sync.js [hunk 2]
```diff
@@ -513,7 +513,11 @@ SyncApi.prototype._syncFromCache = async function(savedSync) {
     debuglog("sync(): not doing HTTP hit, instead returning stored /sync data");
 
     // No previous sync, set old token to null
-    const oldSyncToken = null;
+    const syncEventData = {
+        oldSyncToken: null,
+        nextSyncToken,
```
Comment (richvdh) [inline src/sync.js]: nextSyncToken isn't even set here

Comment (richvdh) [reply src/sync.js]: [I feel like you could have spotted this yourself with a little more care...]

Comment (lukebarnard1) [reply src/sync.js]: oops, sorry

**Pull request merged**